src/Discovery/DiscoveryActionBar.tsx

```diff
@@ -27,8 +29,117 @@ interface Props {
   filtersVisible: boolean;
   setFiltersVisible: (boolean) => void;
   user: User,
+  downloadInProgress: boolean,
+  setDownloadInProgress: (boolean) => void,
+  downloadStatusMessage: {
+    title: string,
+    message: string,
+    active: boolean
+  },
+  setDownloadStatusMessage: (Object) => void;
 }
 
+const handleDownloadZipClick = async (
+  selectedResources: any[],
+  setDownloadInProgress: (boolean) => void,
+  setDownloadStatusMessage: (object) => void,
+  manifestFieldName: string,
+  maxDownloadSizeBytes: number,
+) => {
+  const studyIDs = selectedResources.map((study) => study.project_number);
+  let downloadSize = 0;
+  selectedResources.forEach(
+    (study) => {
+      study[manifestFieldName].forEach(
+        (file) => {
+          downloadSize += file.file_size;
+        },
+      );
+    },
+  );
+
+  if (downloadSize > maxDownloadSizeBytes) {
+    const maxSizeMB = (maxDownloadSizeBytes / 1000000).toFixed(1);
+    const downloadSizeMb = (downloadSize / 1000000).toFixed(1);
+    setDownloadStatusMessage(
+      {
+        title: 'Download limit exceeded',
+        message: `
+        The selected studies contain ${downloadSizeMb} MB of data, which exceeds the download limit of ${maxSizeMB} MB.
+        Please deselect some studies and try again, or use the gen3 client.
+        `,
+        active: true,
+      },
+    );
+    return;
+  }
+
+  setDownloadInProgress(true);
+  setDownloadStatusMessage(
+    {
+      title: 'Your download is being prepared',
+      message: 'Please remain on this page while your download is being prepared.\n\n'
+                 + 'When your download is ready, it will begin automatically. You can close this window.',
+      active: true,
+    },
+  );
+
+  const triggerDownloadResponse = await fetch(
+    '/job/dispatch',
+    {
+      method: 'POST',
+      body: JSON.stringify({
+        action: 'batch-export',
+        input: {
+          study_ids: studyIDs,
+        },
+      }),
+    },
+  );
+  const initialialJobState = await triggerDownloadResponse.json();
+  const downloadFailedMessage = {
+    title: 'Download failed',
+    message: 'There was a problem preparing your download.'
+               + 'Please consider using the gen3 client to download these files via a manifest.',
+    active: true,
+  };
+
+  if (initialialJobState === null) {
```

Comment:
You should consider rewriting this and above section using `fetchWithCreds().then().catch()` chain, for better error handling because you will have HTTP ststus code.

For example the failure of sower job dispatching is not necessarily that sower does not have this said job configured, it could also because user don't have permission to use sower according to user.yaml

If you need a reference, see: https://github.com/uc-cdis/data-portal/blob/master/src/Indexing/Indexing.jsx#L163